src/Auth/EnvMomentoTokenProvider.php

```diff
@@ -27,42 +23,17 @@ public function __construct(
         if ($authToken === false || isNullOrEmpty($authToken)) {
             throw new InvalidArgumentError("Environment variable $envVariableName is empty or null.");
         }
-        if ($trustedControlEndpointCertificateName xor $trustedCacheEndpointCertificateName) {
-            throw new InvalidArgumentError(
-                "If either of trustedCacheEndpointCertificateName or trustedControlEndpointCertificateName " .
-                "are provided, they must both be."
-            );
-        }
-        $this->authToken = $authToken;
-        $payload = AuthUtils::parseAuthToken($authToken);
-        $this->controlEndpoint = $controlEndpoint ?? $payload->cp;
-        $this->cacheEndpoint = $cacheEndpoint ?? $payload->c;
-        $this->trustedControlEndpointCertificateName = $trustedControlEndpointCertificateName;
-        $this->trustedCacheEndpointCertificateName = $trustedCacheEndpointCertificateName;
-    }
-
-    public function getAuthToken(): string
-    {
-        return $this->authToken;
-    }
-
-    public function getCacheEndpoint(): string
-    {
-        return $this->cacheEndpoint;
-    }
-
-    public function getControlEndpoint(): string
-    {
-        return $this->controlEndpoint;
-    }
-
-    public function getTrustedControlEndpointCertificateName(): string|null
-    {
-        return $this->trustedControlEndpointCertificateName;
+        parent::__construct($authToken, $controlEndpoint, $cacheEndpoint, $trustedControlEndpointCertificateName, $trustedCacheEndpointCertificateName);
     }
 
-    public function getTrustedCacheEndpointCertificateName(): string|null
+    /**
+     * Convenience method for reading and parsing a JWT token stored as an environment variable.
+     * @param string $envVariableName Name of the environment variable that contains the JWT token.
+     * @return EnvMomentoTokenProvider
+     * @throws InvalidArgumentError
+     */
+    public static function fromEnvironmentVariable(string $envVariableName): EnvMomentoTokenProvider
```

Comment:
This function should be in the `CredentialProvider` class and should return `ICredentialProvider` type.